### PR TITLE
[Merged by Bors] - update wireframe rendering to new renderer

### PR DESCRIPTION
**Pull request by jakobhellermann**
Updates the wireframe rendering initialliy implemented in https://github.com/bevyengine/bevy/pull/562 to the new renderer.
It lives in `bevy_pbr2` instead of `bevy_render2` because that way it can reuse the `MeshPipeline`.

### CODE REVIEW

File: pipelined/bevy_pbr2/src/wireframe.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    fn specialize(
+        &self,
+        key: Self::Key,
+    ) -> bevy_render2::render_resource::RenderPipelineDescriptor {
+        let mut descriptor = self.mesh_pipeline.specialize(key);
+        descriptor.vertex.shader = self.shader.clone_weak();
+        descriptor.fragment.as_mut().unwrap().shader = self.shader.clone_weak();
+        descriptor.primitive.polygon_mode = PolygonMode::Line;
+        descriptor.depth_stencil.as_mut().unwrap().bias.slope_scale = 1.0;
```
Comment (jakobhellermann) [inline pipelined/bevy_pbr2/src/wireframe.rs:91]: I have no idea if `1.0` is a good value here but it looks way better than without.

grafik
grafik

**Comment by jakobhellermann** [comment]
This isn't actually usable right now because the `POLYGON_MODE_LINE` feature needs to be enabled, but it works once the wgpu feature selection is ported as well.

**Comment by Neo-Zhixing** [comment]
I would recommend having wireframe in `bevy_render2` instead of `bevy_pbr2` because one might want to use `bevy_render2` without pbr. Creating a new pipeline instead of reusing `MeshPipeline` in `bevy_pbr2` would be a better option imo.

**Comment by jakobhellermann** [comment]
> @jakobhellermann should this be updated now that #3282 has been merged, exposing `WgpuOptions`?

The PR is unblocked in that regard, but I'm not sure it can reasonably do anything with the `WgpuOptions`. Since the `WireframePlugin` is added after the `RenderPlugin` it cannot affect the creation of the renderer features.

The old implementation doesn't solve this and requires the user to enable the `POLYGON_MODE_LINE` feature themselves.

**Comment by jakobhellermann** [comment]
The other remaining issue is whether this should live in `render2` or `pbr2`. I agree with @Neo-Zhixing that `render2` is the better location, but to do that properly we need something like https://github.com/bevyengine/bevy/pull/3120 because otherwise we don't know where in the vertex buffer the position attribute is.

**Comment by cart** [comment]
Yeah I agree that `render2` is probably the best place for it and #3120 might help.  I see no real harm in merging as-is though. We can try to move it later after we sort out the vertex attribute situation.

**Pull request closed without merging**